lib/valueflow.cpp

```diff
@@ -1842,6 +1884,7 @@ void ValueFlow::setValues(TokenList *tokenlist, SymbolDatabase* symboldatabase,
     valueFlowPointerAlias(tokenlist);
     valueFlowFunctionReturn(tokenlist, errorLogger, settings);
     valueFlowBitAnd(tokenlist);
+    valueFlowAfterAssign(tokenlist, symboldatabase, errorLogger, settings);
```

Comment:
Unfortunately yes, for some reason it is needed to call this function, because if it is not called, unit tests fails for "for" cycle. I think the code for "for" cycle sets some variable and it is needed to call valueFlowAfterAssign once again.